pt/lessons/libraries/benchee.md

```diff
@@ -0,0 +1,333 @@
+* **median** - quando todos tempos medidos são ordenados, este é o valor médio (ou média dos dois valores do meio quando o número de amostras é par). Devido à inconsistências de ambiente este será mais estável do que a `average`, e um pouco mais provável que reflita a performance normal do seu código em produção. Para esta métrica, um número baixo é melhor.
+
+Há também outras estatísticas disponíveis, mas estas quatro são frequentemente as mais úteis e comumente usadas para _benchmarking_,  por isso elas são exibidas no formatador padrão. Para aprender mais sobre outras métricas disponíveis, confira a documentação [hexdocs](https://hexdocs.pm/benchee/Benchee.Statistics.html#statistics/1).
+
+# Configuração
+
+Uma das melhores partes do Benchee são todas as opções de configuração disponíveis. Examinaremos o básico primeiro, uma vez que não requerem exemplos de código, e então mostraremos como usar uma das melhores características do Benchee - _inputs_.
+
+## Básico
+
+Benchee possui uma grande variedade de opções de configuração. Na interface mais comum `Benchee.run/2`, estes são passados como segundo argumento na forma de uma _keywork list_ opcional:
+
+```elixir
+Benchee.run(%{"example function" => fn -> "hi!" end}, [
+  warmup: 4,
+  time: 10,
+  inputs: nil,
+  parallel: 1,
+  formatters: [&Benchee.Formatters.Console.output/1],
+  print: [ 
+    benchmarking: true,
+    configuration: true,
+    fast_warning: true
+  ],
+  console: [
+    comparison: true,
+    unit_scaling: :best
+  ]
+])
+```
+
+As opções disponíveis são as seguintes (também documentadas em [hexdocs](https://hexdocs.pm/benchee/Benchee.Configuration.html#init/1)).
+
+* **warmup** - o tempo em segundos para o qual um cenário de _benchmarking_ deve ser executado sem tempos de medição antes do início das medidas reais. Isso simula um sistema de funcionamento "quente". Padrão é 2.
+* **time** - o tempo em segundos por quanto tempo cada cenário de _benchmarking_ individual deve ser executado e medido. Padrão é 5.
+* **inputs** - um mapa com _strings_ que representam o nome de entrada como as chaves e a entrada real como valores. Padrão é `nil`. Vamos abordá-lo em detalhes na próxima seção.
+* **parallel** - o número de processos para usar no _benchmark_ de suas funções. Então, se você definir `parallel: 4`, serão gerados 4 processos que executam a mesma função para determinado `time`. Quando estes terminam, então 4 novos processos serão gerados para a próxima função. Isso lhe dá mais dados no mesmo tempo, mas também adiciona mais carga ao sistema interferindo nos resultados do _benchmark_. Isso pode ser útil para simular um sistema sobre carga o que algumas vezes é útil, mas deve ser usado com algum cuidado pois isso pode afetar os resultados de maneiras imprevisíveis. Padrão é 1 (o que significa nenhuma execução em paralelo). 
```

Comment:
What do you think about "under load": it could be better translated to "... para simular um sistema sobrecarregado..."?